common/txmgr/confirmer.go

```diff
@@ -297,7 +297,11 @@ func (ec *Confirmer[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) pro
 		return fmt.Errorf("CheckConfirmedMissingReceipt failed: %w", err)
 	}
 
-	if err := ec.CheckForReceipts(ctx, head.BlockNumber()); err != nil {
+	if head.LatestFinalizedHead() == nil {
```

Comment:
Safe to assume that we will always get a finalized head at some point and we should only operate once we have it?

Reply:
This comment may be relevant here: https://github.com/smartcontractkit/chainlink/pull/13638#discussion_r1672522306

Reply:
Dimitri's [answer](https://chainlink-core.slack.com/archives/D0763PW3UJJ/p1721234907363619)